Trading/Exchange/bybit.py

```diff
@@ -77,48 +82,66 @@ def set_sandbox_mode(self, is_sandboxed):
-            if ExchangeConstantsPositionColumns.MARK_PRICE.value not in position:
-                position[ExchangeConstantsPositionColumns.MARK_PRICE.value] = None  # TODO
-            if ExchangeConstantsPositionColumns.UNREALISED_PNL.value not in position \
-                    and self.BYBIT_UNREALISED_PNL in position:
-                position[ExchangeConstantsPositionColumns.UNREALISED_PNL.value] = position[self.BYBIT_UNREALISED_PNL]
+            if ExchangeConstantsPositionColumns.TIMESTAMP.value not in position_dict \
+                    and self.BYBIT_TIMESTAMP in position_dict:
+                position_dict[ExchangeConstantsPositionColumns.TIMESTAMP.value] = position_dict[self.BYBIT_TIMESTAMP]
+            if ExchangeConstantsPositionColumns.MARK_PRICE.value not in position_dict:
+                position_dict[ExchangeConstantsPositionColumns.MARK_PRICE.value] = None  # TODO
+            if ExchangeConstantsPositionColumns.UNREALISED_PNL.value not in position_dict \
+                    and self.BYBIT_UNREALISED_PNL in position_dict:
+                position_dict[ExchangeConstantsPositionColumns.UNREALISED_PNL.value] = position_dict[
+                    self.BYBIT_UNREALISED_PNL]
         except KeyError as e:
             self.logger.error(f"Fail to cleanup position dict ({e})")
-        return position if position[self.BYBIT_SIZE] != 0 else None
+        return position_dict if position_dict[self.BYBIT_SIZE] != 0 else None
+
+    def cleanup_funding_dict(self, funding_dict, from_ticker=False):
+        old_funding_dict = {}
+        if from_ticker and CCXT_INFO in funding_dict:
+            funding_dict, old_funding_dict = funding_dict[CCXT_INFO], funding_dict
 
-    def _cleanup_funding_dict(self, funding_dict):
         try:
             if ExchangeConstantsFundingColumns.TIMESTAMP.value not in funding_dict \
                     and self.BYBIT_FUNDING_TIMESTAMP in funding_dict:
                 funding_dict[ExchangeConstantsFundingColumns.TIMESTAMP.value] = funding_dict[
                     self.BYBIT_FUNDING_TIMESTAMP]
 
+            # if still empty use old_funding_dict if available
+            if old_funding_dict and ExchangeConstantsFundingColumns.TIMESTAMP.value not in funding_dict \
+                    and CCXT_INFO in old_funding_dict:
+                funding_dict[ExchangeConstantsFundingColumns.TIMESTAMP.value] = old_funding_dict[CCXT_INFO]
+
             if ExchangeConstantsFundingColumns.NEXT_FUNDING_TIME.value not in funding_dict:
                 funding_dict[ExchangeConstantsFundingColumns.NEXT_FUNDING_TIME.value] = \
-                    funding_dict[ExchangeConstantsFundingColumns.TIMESTAMP.value] + self.BYBIT_DEFAULT_FUNDING_TIME # TODO manage timezone
+                    funding_dict[
+                        ExchangeConstantsFundingColumns.TIMESTAMP.value] + self.BYBIT_DEFAULT_FUNDING_TIME  # TODO manage timezone
```

Comment:
I don't understand this

Reply:
You mean the `TODO manage timezone` or the calculation ?

Reply:
Yes, this and ```+ self.BYBIT_DEFAULT_FUNDING_TIME```, why can't we use the exchange data directly ?